### PR TITLE
Decorate

### PR DESCRIPTION
**Pull request by VitorGuedesMadeira**
### OOP school library: decorate a class
**_✅ Create a class Nameable._**
 - Implement a method called correct_name that will raise a NotImplementedError.

**_✅ Turn Person class to Nameable_**
 - Make sure that your Person class inherits from Nameable
 - Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

**_✅ Prepare base Decorator_**
 - Make sure that it inherits from Nameable.
 - In the constructor assign a nameable object from params to an instance variable.
 - Implement the correct_name method that returns the result of the correct_name method of the @nameable.

**_✅ Add CapitalizeDecorator:_**
 - Create a class that inherits from the base Decorator class.
 - Implement a method correct_name that capitalizes the output of @nameable.correct_name.

**_✅ Add TrimmerDecorator:_**
 - Create a class that inherits from the base Decorator class.
 - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.
 
gTN

### CODE REVIEW

File: nameable_class.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
```
Comment (RolandM99) [inline nameable_class.rb]: - [ ] `Nameable` class should look as below:
```ruby
class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end
```
Kindly, make sure that you add the missing syntax in your code.

File: decorator_trimmer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator_base'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0..9]
+  end
+end
```
Comment (RolandM99) [inline decorator_trimmer.rb]: - [ ] the name should have a maximum of 10 characters. Kindly, make sure that your TrimeDecorator class look as the snippet code below to ensure that you meet the requirement:
```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```
Make sure that you complete the missing syntax!

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi @VitorGuedesMadeira  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @VitorGuedesMadeira, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**